packages/algorand-js/src/interpreter/opcode-list.ts

```diff
@@ -729,22 +727,20 @@ export class NotEqualTo extends Op {
     this.assertMinStackLen(stack, 2);
     const last = stack.pop();
     const prev = stack.pop();
-    if (typeof last === typeof prev) {
-      if (typeof last === "bigint") {
-        if (last === prev) {
-          stack.push(BIGINT0);
-        } else {
-          stack.push(BIGINT1);
-        }
+    if (typeof last !== typeof prev) {
+      throw new TealError(ERRORS.TEAL.INVALID_TYPE);
+    } else if (typeof last === "bigint") {
+      if (last === prev) {
+        stack.push(BIGINT0);
       } else {
-        if (compareArray(this.assertBytes(last), this.assertBytes(prev))) {
-          stack.push(BIGINT0);
-        } else {
-          stack.push(BIGINT1);
-        }
+        stack.push(BIGINT1);
       }
     } else {
-      throw new TealError(ERRORS.TEAL.INVALID_TYPE);
+      if (compareArray(this.assertBytes(last), this.assertBytes(prev))) {
+        stack.push(BIGINT0);
+      } else {
+        stack.push(BIGINT1);
+      }
```

Comment:
this thing is repeated many times. Please move it to function:
```
function pushBolleanCheck(stack, ok: boolean) {
  if (ok) 
    stack.push(BIGINT0)
  else ...
}

Reply:
BTW, are you sure we should push `0` when things are equal?

Reply:
yes, because here opcode is `notEqualTo`.